### ISSUE TITLE
`validate_json` returns `syntax_error` if buffer contains non-ASCII chars

### ISSUE THREAD

**Issue by Chaoses-Ib** [stephenberry/glaze#977]
Example:
```cpp
// In ANSI (GBK) encoding
auto e = glz::validate_json(R"({"key":"value中文"})");

// In UTF-8
auto e = glz::validate_json((const char*)u8R"({"key":"value中文"})");
```
![image](https://github.com/stephenberry/glaze/assets/50246090/56697b10-b8f6-44d5-9d9c-f1695471dd15)

![image](https://github.com/stephenberry/glaze/assets/50246090/eff04b20-16f5-41e6-9dcc-3b3dfd58dc60)

According to [RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259), "JSON text exchanged between systems that are not part of a closed ecosystem MUST be encoded using UTF-8". I think at least buffers in UTF-8 encoding should be accepted.

Version: 2.6.0

**Comment by stephenberry** [comment]
Thanks so much for reporting this. Will plan to fix soon.

**Comment by stephenberry** [comment]
Thanks again for reporting this. I've merged a fix (#979). I wasn't taking the sign-ness of `char` into account in some places.

**Comment by Chaoses-Ib** [comment]
Thanks for the quick fix. I've made a PR to update the version used by vcpkg: https://github.com/microsoft/vcpkg/pull/38592
